CMSgov/mint-app
# [EASI-4195] Native browser prompt on app leave

# EASI-4195


## Changes and Description

This PR covers scenarios where uses may want to save data on tab close, refresh, etc.  This requires native browser support. We cannot create custom component for this, only text

TODO:  Natasha is formulating the text for this prompt

- Created `useConfirmPageLeave` hook to trigger the browser prompt
- Create util `ConfirmLeave`  for incorporating Formik context 's `dirty` state, to indicate if prompt should be rendered
- Implemented `<ConfirmLeave />` on all forms that previously had autosave
- Added a few spots where mutation handler was missing



## How to test this change

- Navigate to nay form page
- Touch/fill out a field
- Attempt to refresh, close tab, etc
- Verify prompt appears and allows you to stay if selected

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
